cmd/tuf/app/init.go

```diff
@@ -144,12 +144,26 @@ func InitCmd(ctx context.Context, directory string,
 		return err
 	}
 
+	// This initializes a new repository, creating a new root.json.
 	if err := repo.Init(ConsistentSnapshot); err != nil {
 		// If there was already a repository present, then this will fail on
 		// ErrInitNotAllowed. Allow this to happen for chaining repositories.
 		if !errors.Is(err, tuf.ErrInitNotAllowed) {
 			return err
 		}
+		// If there is already a repository present, force a root refresh by
+		// definitely updating the threshold.
+		curThreshold, err := repo.GetThreshold("root")
+		if err != nil {
+			return err
+		}
+		// The risk is mitigated here by increasing the threshold.
+		// This code does not change the threshold, but simply stages the new
+		// root. A fix like https://github.com/theupdateframework/go-tuf/issues/239
+		// to bump expiration would help.
```

Comment:
Can you explain more why this does not change the threshold and only stages a new root?

Reply:
Ah, I should have been more clear - I'll update the comment. It's because we always set the correct threshold below https://github.com/sigstore/root-signing/blob/41c1c0aaea1ddb8cf950bf3ec96dffca0ae62774/cmd/tuf/app/init.go#L167

But when that call is a no-op, it does NOT stage any new metadata.

Reply:
Ah, of course! Thanks for clarifying.

Reply:
You know, upon sleeping on this, I'll change this internally in `setMetaWithSigKeyIDs` - I'll make it `SetMeta` on the staged keys and THEN add the key IDs.

Reply:
See new commit! Thank you for asking this question and pushing me to address this :) The change is a lot better now.